subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -517,6 +527,13 @@ static int hawkbit_parse_deployment(struct hawkbit_dep_res *res,
 	}
 
 	artifact = &chunk->artifacts[0];
+	hash_info = mbedtls_md_info_from_type(MBEDTLS_MD_SHA256);
+	if (hex2bin(artifact->hashes.sha256, mbedtls_md_get_size(hash_info),
```

Comment:
This is wrong, second arg of `hex2bin` is the length of the hex string provided, so it should be 64 here instead of 32.
The `mbedtls_md_get_size(hash_info)` has to be replaced by either `strlen(artifact->hashes.sha256)` or a hardcoded `#define` macro.

@sylvioalves What do you think?

Reply:
I am not exactly aware of what is provided by the hawkbit server. However, I agree it won't work as the hex-string would be then 64 bytes instead of 32 as from the byte-array hash.  I am ok then using previous modification.

Reply:
@sylvioalves I will use the `hex2bin` instead of the previously custom conversion function, just that I'd have to use `strlen` instead of `strnlen` because `strnlen` requires a max length arg and if I use `strnlen`, I might as well just use a defined macro. And you raised some security concert with `strlen` last time, that's why I'd like to have your comments.

FYI it will receive this:

```json
{
    "id": "2",
    "deployment": {
        "download": "forced",
        "update": "forced",
        "chunks": [
            {
                "part": "bApp",
                "version": "0.0.1",
                "name": "hawkbit",
                "artifacts": [
                    {
                        "filename": "zephyr.signed.bin",
                        "hashes": {
                            "sha1": "ffe8f293999df74153b53ac53618d65b1e3f6764",
                            "md5": "23d17ad9288c229941d0806e4d78af03",
                            "sha256": "95141b6b8a70676d204bbc4587666a691f8cc651e43596824a2d05e360d3da41"
                        },
                        "size": 205765,
                        "_links": {
                            "download-http": {
                                "href": "http://123.123.12:9010/DEFAULT/controller/v1/ivm22_g0b1re-2037334b3436500b004d003f/softwaremodules/1/artifacts/zephyr.signed.bin"
                            },
                            "md5sum-http": {
                                "href": "http://123.123.12:9010/DEFAULT/controller/v1/ivm22_g0b1re-2037334b3436500b004d003f/softwaremodules/1/artifacts/zephyr.signed.bin.MD5SUM"
                            }
                        }
                    }
                ]
            }
        ]
    }
}
```

Reply:
I don't see a problem having the some  `#define HAWKBIT_HASH_STR_SIZE 64` for this. Any other sugestion?